### PR TITLE
feat(model): add support to upload files

### LINKED ISSUE #83: Upload a PDF file
Hello there,

I am trying to use this awesome plugin to upload a PDF file. I have created a model for this:

```js
import Model from './Model';

export default class JobApplication extends Model {
  resource() {
    return 'v2/job_applications';
  }
}
```
Then, I call the model within the `.vue` file like that:

```js
...
 import JobApplication from "./../models/JobApplication";
...
async onSubmit() {
  let job = new JobApplication({
    first_name,
    last_name,
    email,
    mobile_phone,
    resume: this.$refs.resume.files[0]
  });
  await job
    .save()
    .then(response => {
      console.log(response);
    })
    .catch(error => {
      console.error(error.response);
    });
}
...
```
The problem is that the POST request fails every time I try to submit the form because of the resume field. What am I doing wrong in here? Do I have to do any extra steps? Thank you

### PR DESCRIPTION
**Pull request by JoaoPedroAS51**
Add support to upload files by checking if data has instance of `File`. If so, it set the `Content-Type` to `multipart/form-data` and convert the data object to `FormData` using [object-to-formdata](https://www.npmjs.com/package/object-to-formdata).

Require #142 

Based on [comment](https://github.com/robsontenorio/vue-api-query/issues/83#issuecomment-614743771) by @alvaro-canepa at #83 

Closes #83 

### TODO

- [x] Update documentation
- [x] Update tests
- [x] Add dependency `object-to-formdata`

### CODE REVIEW

**Comment by robsontenorio** [comment]
Beautiful!

I know there are some gotchas behind FormData, I can’t remember right know. But, it is a optional feature and there are no breaking changes, looks good for me. Let’s try it. 

Please, wait reviews from #142

File: src/utils.js [hunk 1]
```diff
@@ -44,3 +44,101 @@ export function setProp (holder, propName, value) {
...
+  value &&
+  typeof value.size === 'number' &&
+  typeof value.type === 'string' &&
+  typeof value.slice === 'function';
+
+const isFile = (value) =>
+  isBlob(value) &&
+  typeof value.name === 'string' &&
+  (typeof value.lastModifiedDate === 'object' ||
+    typeof value.lastModified === 'number')
```
Comment (robsontenorio) [inline src/utils.js:144]: Don’t we have some package to do this hard work for us?

File: src/Model.js [hunk 2]
```diff
@@ -47,6 +78,10 @@ export default class Model extends StaticModel {
     return this[this.primaryKey()]
   }
 
+  updateMethod() {
+    return 'PUT'
+  }
```
Comment (robsontenorio) [inline src/Model.js:83]: As discussed, this should be removed , in favor of new config() method.

File: src/utils.js (hunk 1 above)
Comment (robsontenorio) [reply src/utils.js:144]: I mean, replace all this file with some package.

File: src/Model.js (hunk 2 above)
Comment (JoaoPedroAS51) [reply src/Model.js:83]: Yes, it's on TODO list.

File: src/utils.js (hunk 1 above)
Comment (JoaoPedroAS51) [reply src/utils.js:144]: Yeah, I agree. I just wasn't sure if you would like to have dependencies, so I copied the src, but if there is no problem with that, I can use the package.

File: src/Model.js (hunk 2 above)
Comment (JoaoPedroAS51) [reply src/Model.js:83]: This PR requires the #142. The `updateMethod` will be removed there.

File: src/utils.js (hunk 1 above)
Comment (JoaoPedroAS51) [reply src/utils.js]: Done :)

Comment (JoaoPedroAS51) [reply src/utils.js]: > I mean, replace all this file with some package.

Ok, then I will add those packages, but in a separate PR.

**Review by robsontenorio — APPROVED** [review]

**Comment by Peter-Krebs** [comment]
Thanks for taking the time to implement @JoaoPedroAS51 that was a long awaited feature 🚀

**Pull request merged**